### PR TITLE
DOC: Fix consistent typo in contrastive loss

### PR DESCRIPTION
**Pull request by gustavla**
If a pair is similar, the contrastive loss should take the squared distance and not the distance. This is clearly what the code is doing, but the documentation gets it consistently wrong.

This is straight from the code with an arrow added:

```cpp
if (static_cast<int>(bottom[2]->cpu_data()[i])) {  // similar pairs
  loss += dist_sq_.cpu_data()[i];  // <---- clearly d^2 and not d
} else {  // dissimilar pairs
  if (legacy_version) {
    loss += std::max(margin - dist_sq_.cpu_data()[i], Dtype(0.0));
  } else {
    Dtype dist = std::max(margin - sqrt(dist_sq_.cpu_data()[i]), 0.0);
    loss += dist*dist;
  }
}
```

### CODE REVIEW

**Comment by seanbell** [comment]
LGTM

**Comment by shelhamer** [comment]
Thanks for the doc fix @gustavla!
